File: mybatis-plus-join-test/test-join/src/main/java/com/github/yulichang/test/join/entity/UserDto.java
package com.github.yulichang.test.join.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.github.yulichang.annotation.Table;
import lombok.Data;

import java.io.Serializable;

@Table
@Data
@TableName("user_dto")
public class UserDto implements Serializable {

    @TableId
    private Integer id;

    private Integer userId;

    @TableField(exist = false)
    private String userName;

    private Integer createBy;

    @TableField(exist = false)
    private String createName;

    private Integer updateBy;

    @TableField(exist = false)
    private String updateName;
}

File: mybatis-plus-join-test/test-kotlin/src/main/kotlin/com/github/yulichang/test/kt/entity/UserDto.kt
package com.github.yulichang.test.kt.entity

import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableName
import java.io.Serializable

@TableName("user_dto")
class UserDto : Serializable {
    @TableId
    val id: Int? = null
    val userId: Int? = null

    @TableField(exist = false)
    val userName: String? = null
    val createBy: Int? = null

    @TableField(exist = false)
    val createName: String? = null
    val updateBy: Int? = null

    @TableField(exist = false)
    val updateName: String? = null
}
